Shorten compilation time of two MatrixFree tests

kronbichler opened the pull request:
Some hp tests of the matrix-free framework were plagued by long compile times when using templated polynomial degrees to generate all code. It is better to let the code run through the pre-compiled paths, i.e., selecting `-1` as template argument.

Related to #15519.

peterrum commented on tests/matrix_free/matrix_vector_hp.cc:
```diff
@@ -45,50 +45,11 @@ class MatrixFreeTestHP
-                                                    src,
-                                                    subrange_deg);
+    if (data.get_cell_active_fe_index(cell_range) > 0) // exclude FE_Nothing
+      helmholtz_operator<dim, -1, Vector<Number>, 0>(data,
```
I would prefer keep the test as it is to check if the template parameters are still picked up correctly in the hp case. We already have a non-templated test: https://github.com/dealii/dealii/blob/3bba74760bff6724e1766bf22e4ee06eaf98bfc1/tests/matrix_free/matrix_vector_hp_no_template.cc#L56C9-L56C39, which is, as far as I remember correctly, a copy of the current test.

kronbichler replied:
But we should then simplify the test to only run in 2D and only use 2 or at most 3 polynomial degrees that get their code compiled. If you agree, I can make that change.

peterrum commented:
> If you agree, I can make that change.

:+1:

kronbichler commented:
Alright, I changed the `matrix_vector_hp`, which is the most comprehensive test in terms of detection, back to the template deduction. At the same time, I now also switched the `matrix_free_hp_parallel_01` tests to the `-1` template because the deduction mechanism is tested in the non-MPI version, and the MPI case is supposed to stress other parts of the algorithms.

kronbichler commented:
/rebuild
